### PR TITLE
TEX-310 Remove aws sdk core, replace with cloudwatch and bump version to 0.5.3

### PR DESCRIPTION
**Pull request by alexdengoptimal**

### CODE REVIEW

**Review by bduran82 — COMMENTED** [review]
Should we fix the version to avoid breaking this gem if the aws gem upgrades and is not backwards compatible?

**Comment by lucy-dement** [comment]
> Should we fix the version to avoid breaking this gem if the aws gem upgrades and is not backwards compatible?

Given that there's no gemfile.lock to stop un-intended upgrades it's probably a good idea.

File: cloudwatch-metrics-resque.gemspec [hunk 1]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.4'
 
-  spec.add_dependency 'aws-sdk-core', '~> 3'
```
Comment (lucy-dement) [inline cloudwatch-metrics-resque.gemspec:26]: We require this in `lib/cloud_watch_metrics/util.rb` is that going to break?

Comment (alexdengoptimal) [reply cloudwatch-metrics-resque.gemspec:26]: Ahh crap I didn't check the file with the change in. Will check that in now

File: cloudwatch-metrics-resque.gemspec [hunk 2]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |spec|
...
+  spec.add_dependency 'aws-sdk-cloudwatch', '1.58.0'
```
Comment (lucy-dement) [inline cloudwatch-metrics-resque.gemspec]: Can you lock this to the major version not the minor ones?

**Review by lucy-dement — APPROVED** [review]
And now we get to test in spacewaffle 🎉

**Pull request merged**